spec/datadog/core/environment/execution_spec.rb

```diff
@@ -178,6 +169,15 @@ def test_it_does_something_useful
 
         let(:script) do
           <<-'RUBY'
+            # Under Ruby 3.0 through 3.2 there is a weird error that occurs
+            # in CI where two copies of psych get loaded in the same process,
+            # and even more strangely the first version is a newer one from
+            # gem and the second one is the older one from Ruby standard
+            # library. Try to work around this situation by forcing psych
+            # to be loaded from (some) gem.
+            # We still don't know exactly what is causing the original issue.
+            gem 'psych'
```

Comment:
I'm somewhat curious about the need for this change? E.g. why is it needed now here but wasn't before; and why doesn't the place where it was needed previously still need it?

Reply:
The commit is cherry-picked from a merged branch that address a flaky test to see if it improves